agents/agent.py

```diff
@@ -226,21 +296,44 @@ def to_dict(self):
             'healthy': self.service_healthy(),
         }
 
-    def to_json(self):
-        """Dump host metadata to json file."""
+    def status_to_json(self, log=False):
+        """
+        Dump host metadata to json file.
+
+        Parameters:
+            log (bool): Whether to log JSON status string to the logfile.
+                Default is False.
+
+        Returns:
+            str: JSON status string.
+        """
         try:
-            msg = json.dumps(self.to_dict())
-            self.logger.info(msg)
-        except (IOError, OSError) as e:
+            status = json.dumps(self.status_to_dict())
```

Comment:
If any value in `status_to_dict()` is not JSON-serializable (e.g., `datetime`), it will fail.
Use `default=str` in `json.dumps()`.

Reply:
Fixed.